src/middlewared/middlewared/plugins/docker/state_management.py

```diff
@@ -107,8 +107,19 @@ async def _event_system_ready(middleware, event_type, args):
     if await middleware.call('failover.licensed'):
         return
 
-    if (await middleware.call('docker.config'))['pool']:
-        middleware.create_task(middleware.call('docker.state.start_service'))
+    docker_pool_configured = bool((await middleware.call('docker.config'))['pool'])
+    start_docker = True
+    if (
+        (await middleware.call('docker.config'))['nvidia'] and
+        await middleware.call('nvidia.present') and
+        not await middleware.call('nvidia.installed')
+    ):
+        await middleware.call('nvidia.install', docker_pool_configured)
```

Comment:
Should we wait for this to complete since it's a job?

Reply:
Hmm, so it seems you'll start a background task in this event and not start the docker service here but instead start it after the installation has completed. I guess that makes sense.